src/acl/Checklist.cc

```diff
@@ -137,8 +136,8 @@ ACLChecklist::goAsync(AsyncState *state)
     ++asyncLoopDepth_;
 
     asyncStage_ = asyncStarting;
-    changeState(state);
-    state->checkForAsync(this); // this is supposed to go async
+    changeState(&starter);
+    starter(*this, acl); // this is supposed to go async
 
     // Did AsyncState object actually go async? If not, tell the caller.
```

Comment:
```suggestion
    // Did starter() actually go async? If not, tell the caller.
```

Please check for other stale AsyncState references that are not just type renaming (that renaming is a known PR TODO; if you feel it is best to implement that TODO now because it does not reduce/complicate PR diff by much, then let's do that now).